src/tests/daos_racer.c

```diff
@@ -372,6 +372,45 @@ racer_test_idx(struct racer_sub_tests *tests)
 	return idx;
 }
 
+static bool
+racer_valid_oid(daos_obj_id_t oid, daos_pool_info_t *pinfo)
+{
+	daos_oclass_id_t	ocid;
+	int			required_node;
+	int			required_tgt;
+
+	ocid = daos_obj_id2class(oid);
+	switch (ocid) {
+	case OC_RP_XSF:
+		/* Skip single replicated objects. */
+		return false;
+	case OC_RP_2G1:
+		required_node = 2;
+		required_tgt = 2;
+		break;
+	case OC_RP_2G2:
+		required_node = 2;
+		required_tgt = 4;
+		break;
+	case OC_RP_3G1:
+		required_node = 3;
+		required_tgt = 3;
+		break;
+	case OC_RP_3G2:
+		required_node = 3;
+		required_tgt = 6;
+		break;
+	default:
```

Comment:
hmm, you can use daos_obj_layout_get() or simply use
daos_oclass_attr_find(oid) to get node/tgt information? instead of list all of it?